src/map_object.c

```diff
@@ -107,493 +119,472 @@ LocalMapObject* sub_0205E1D0(MapObjectManager* manager, ObjectEvent* object_even
     MapObject_SetInitialX(object, x);
-    MapObject_SetPrevX(object, x);
+    MapObject_SetPreviousX(object, x);
     MapObject_SetCurrentX(object, x);
 
-    position_vec.y = ObjectEventTemplate_GetHeight(template);
-    u32 height = (position_vec.y >> 3) / FX32_ONE;
+    coords.y = ObjectEvent_GetHeight(objectEvent);
+    u32 height = (coords.y >> 3) / FX32_ONE;
     MapObject_SetInitialHeight(object, height);
-    MapObject_SetPrevHeight(object, height);
+    MapObject_SetPreviousHeight(object, height);
     MapObject_SetCurrentHeight(object, height);
 
-    u16 y = ObjectEventTemplate_GetYCoord(template);
-    position_vec.z = y * (16 * FX32_ONE) + (8 * FX32_ONE);
+    u16 y = ObjectEvent_GetYCoord(objectEvent);
+    coords.z = y * FX32_CONST(16) + FX32_CONST(8);
     MapObject_SetInitialY(object, y);
-    MapObject_SetPrevY(object, y);
+    MapObject_SetPreviousY(object, y);
     MapObject_SetCurrentY(object, y);
 
-    MapObject_SetPositionVec(object, &position_vec);
+    MapObject_SetPositionVec(object, &coords);
 }
 
-void sub_0205EC90(LocalMapObject* object, MapObjectManager* manager) {
-    MapObject_SetFlagsBits(object, MAPOBJECTFLAG_UNK12 | MAPOBJECTFLAG_UNK11 | MAPOBJECTFLAG_ACTIVE);
+static void sub_0205EC90(LocalMapObject *object, MapObjectManager *manager) { //setup facing and flags? also sets manager?
+    MapObject_SetFlagsBits(object, (MapObjectFlagBits)(MAPOBJECTFLAG_UNK12 | MAPOBJECTFLAG_UNK11 | MAPOBJECTFLAG_ACTIVE));
     if (MapObject_ScriptIdIsFFFF(object) == TRUE) {
         MapObject_SetFlag25(object, TRUE);
     }
-    sub_0205F354(object, manager);
-    MapObject_ForceSetFacingDirection(object, MapObject_GetInitialFacing(object));
-    MapObject_SetNextFacing(object, MapObject_GetInitialFacing(object));
+    MapObject_SetManager(object, manager);
+    MapObject_SetFacingDirectionDirect(object, MapObject_GetInitialFacingDirection(object));
+    MapObject_SetNextFacingDirection(object, MapObject_GetInitialFacingDirection(object));
     MapObject_ClearHeldMovement(object);
 }
 
-void sub_0205ECE0(LocalMapObject* object) {
-    u32 movement = MapObject_GetMovement(object);
-    UnkLMOCallbackStruct* unk = sub_0205FB00(movement);
+static void sub_0205ECE0(LocalMapObject *object) {
+    UnkLMOCallbackStruct *unk = sub_0205FB00(MapObject_GetMovement(object));
     sub_0205F414(object, sub_0205FB18(unk));
     sub_0205F428(object, sub_0205FB1C(unk));
     sub_0205F43C(object, sub_0205FB20(unk));
 }
 
 extern const UnkLMOCallbackStruct2 ov01_0220724C;
 
-static void sub_0205ED18(LocalMapObject* object) {
-    u32 gfx_id = MapObject_GetGfxID(object);
-    UnkLMOCallbackStruct2* unk = (gfx_id == 0x2000) ? (UnkLMOCallbackStruct2*)&ov01_0220724C : sub_0205FB38(gfx_id);
+static void sub_0205ED18(LocalMapObject *object) {
+    u32 spriteId = MapObject_GetSpriteID(object);
+    UnkLMOCallbackStruct2 *unk = (spriteId == SPRITE_CAMERA_FOCUS) ? (UnkLMOCallbackStruct2 *)&ov01_0220724C : sub_0205FB38(spriteId);
```

Comment:
This should probably be a const pointer

Reply:
out of scope, will fix in PR 2

Reply:
since it requires many changes to funcs I haven't got to yet in this file